drivers/flash/nrf_qspi_nor.c

```diff
@@ -556,11 +587,30 @@ static int qspi_nor_write(struct device *dev, off_t addr, const void *src,
 		return -EINVAL;
 	}
 
-	qspi_lock(dev);
-
-	nrfx_err_t res = nrfx_qspi_write(src, size, addr);
 
-	qspi_wait_for_completion(dev, res);
+	nrfx_err_t res;
+#if IS_ENABLED(NORDIC_QSPI_NOR_FLASH_ALLOW_STACK_USAGE_FOR_DATA_IN_FLASH)
+	if (((u32_t)src < CONFIG_SRAM_BASE_ADDRESS) && !(size < 4U)) {
+		/* if the value is in flash we need to copy it into RAM since
+		 * the DMA won't be able to transport data from flash to QSPI
+		 */
+		src = buf;
```

Comment:
Can we get rid of `src` within this `if`, `buf` is the source of data here anyway, and it becomes confusing when in 599 there is copy to `buf` done, but in 601 write is done from `src` and between these lines there is no magic that would change `buf` to `src`.

Reply:
Done